exercises/practice/kindergarten-garden/test_kindergarten_garden.c

```diff
@@ -0,0 +1,164 @@
+#include "kindergarten_garden.h"
+
+#include "test-framework/unity.h"
+
+void setUp(void)
+{
+}
+
+void tearDown(void)
+{
+}
+
+static void test_kindergarten_garden(const char *diagram, const char *student,
+                                     const plant_t *expected)
+{
+   plant_t actual[4];
+   plants(diagram, student, actual);
+
+   TEST_ASSERT_EQUAL_INT_ARRAY(expected, actual, 4);
+}
+
+static void test_partial_garden_garden_with_single_student(void)
+{
+   const plant_t expected[] = { RADISHES, CLOVER, GRASS, GRASS };
+   test_kindergarten_garden("RC\nGG", "Alice", expected);
+}
+
+static void test_partial_garden_different_garden_with_single_student(void)
+{
+   const plant_t expected[] = { VIOLETS, CLOVER, RADISHES, CLOVER };
+   test_kindergarten_garden("VC\nRC", "Alice", expected);
+}
+
+static void test_partial_garden_garden_with_two_students(void)
+{
+   const plant_t expected[] = { CLOVER, GRASS, RADISHES, CLOVER };
+   test_kindergarten_garden("VVCG\nVVRC", "Bob", expected);
```

Comment:
You could consider formatting the "diagram" in the tests in two lines like this:
```c
   test_kindergarten_garden(
         "VVCG\n"
         "VVRC",
         "Bob", expected);
```
to make the tests a little bit easier to read.